src/OpenLoco/src/GameCommands/CreateWall.cpp

```diff
@@ -0,0 +1,288 @@
+        {
+            setErrorText(StringIds::error_can_only_build_above_ground);
+            return FAILURE;
+        }
+
+        if ((slopeFlags & (EdgeSlope::upwards | EdgeSlope::downwards)) != EdgeSlope::none)
+        {
+            auto testHeight = targetHeight + kSmallZStep;
+
+            // Test placement edges to ensure we don't build partially underground
+            for (auto i = 2; i <= 3; i++)
+            {
+                auto testEdge = (args.rotation + i) & 3;
+                if (surface->slope() & (1 << testEdge))
+                {
+                    if (args.pos.z < surface->baseZ())
+                    {
+                        setErrorText(StringIds::error_can_only_build_above_ground);
+                        return FAILURE;
+                    }
+
+                    if (surface->isSlopeDoubleHeight())
+                    {
+                        testEdge = (testEdge - 1) & 3;
+                        if (surface->slope() & (1 << testEdge))
+                        {
+                            testEdge = (testEdge + 2) & 3;
+                            if (surface->slope() & (1 << testEdge))
+                            {
+                                if (args.pos.z < testHeight + kSmallZStep)
+                                {
+                                    setErrorText(StringIds::error_can_only_build_above_ground);
+                                    return FAILURE;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        auto targetBaseZ = targetHeight; // / kSmallZStep;
+        auto clearZ = targetBaseZ;
+
+        // TODO: fold into previous block; left for now to match IDA
+        auto* wallObj = ObjectManager::get<WallObject>(args.type);
+        if ((slopeFlags & (EdgeSlope::upwards | EdgeSlope::downwards)) != EdgeSlope::none)
+        {
+            if ((wallObj->flags & WallObjectFlags::onlyOnLevelLand) != WallObjectFlags::none)
+            {
+                setErrorText(StringIds::can_only_build_this_on_level_land);
+                return FAILURE;
+            }
+
+            clearZ += kSmallZStep;
+        }
+
+        if (!canConstructWall(args.pos, targetBaseZ, clearZ, args.rotation))
+        {
+            return FAILURE;
+        }
+
+        if (!TileManager::checkFreeElementsAndReorganise())
+        {
+            // Error message set in checkFreeElementsAndReorganise
+            return FAILURE;
+        }
+
+        if (!(flags & Flags::apply))
+        {
+            return 0;
+        }
+
+        auto* wall = TileManager::insertElement<WallElement>(args.pos, targetBaseZ, 0);
+        if (wall == nullptr)
+        {
+            return FAILURE;
+        }
+
+        wall->setClearZ(clearZ);
+        wall->setRotation(args.rotation); // original had slopeFlags in here as well, but that interferes with type?? check
```

Comment:
yes you will need to have a setSlopeFlags function

Reply:
Is that different from the `setSlope` that happens one line later?

Reply:
Yes.
`wall->setSlopeFlags(slopeFlags);`
```cpp
void setSlopeFlags(uint8_t slopeFlags){
    _type &= ~0xC0;
    _type |= slopeFlags & 0xC0; // Note slope flags are already << 6
}
```